packages/satusehat/src/Utils.js

```diff
@@ -0,0 +1,88 @@
+import { composeNextState } from '@openfn/language-common';
+import {
+  request as commonRequest,
+  logResponse,
+} from '@openfn/language-common/util';
+import pkg from '../package.json'  assert { type: 'json' };
+
+export const authorize = state => {
+  const auth = state.configuration;
+  if (auth.access_token) {
+    return state;
+  }
+  const clientId = auth.clientId;
+  const clientSecret = auth.clientSecret;
+  const headers = {};
+
+  if (clientId && clientSecret) {
+    Object.assign(headers, {
+      'Content-Type': 'application/x-www-form-urlencoded',
+      'User-Agent': `nodejs/${process.version} @openfn/language-satusehat/${pkg.version}`,
+    });
+
+    const body = new URLSearchParams();
+    body.append('client_id', clientId);
+    body.append('client_secret', clientSecret);
+
+    const options = {
+      body: body.toString(),
+      headers,
+      method: 'POST',
+      parseAs: 'json',
+      baseUrl: auth.baseUrl,
+      query: {
+        grant_type: 'client_credentials',
+      },
+      maxRedirections: 1,
+    };
+
+    return commonRequest('POST', '/oauth2/v1/accesstoken', options).then(
+      response => {
+        return {
+          ...state,
+          configuration: {
+            ...state.configuration,
+            access_token: response.body.access_token,
+          },
+        };
+      }
+    );
+  } else {
+    throw new Error(
+      'Invalid authorization credentials. Include clientId and clientSecret in state.configuration'
+    );
+  }
+};
+
+export const prepareNextState = (state, response, callback = s => s) => {
+  const { body, ...responseWithoutBody } = response;
+  const nextState = {
+    ...composeNextState(state, response.body),
+    response: responseWithoutBody,
+  };
+
+  return callback(nextState);
+};
+
+export async function request(configuration, path, opts) {
+  const { baseUrl, access_token } = configuration;
+
+  const { method, data, params = {}, parseAs = 'json' } = opts;
+
+  const headers = {
+    Authorization: `Bearer ${access_token}`,
+    'User-Agent': `nodejs/${process.version} @openfn/language-satusehat`,
```

Comment:
There's a second user-agent string here.

Maybe we can add a single `generateUserAgent` function to handle this for us?